From 8d3e9e5c3e1813b372f84d3bcca08bbe479f7f63 Mon Sep 17 00:00:00 2001
From: Lee Duncan <leeman.duncan@gmail.com>
Date: Tue, 21 May 2013 14:16:38 -0700
Subject: PATCH 1 of 1] correctly check return value of nice()

The nice() call can return a negative value, since
it returns the previous nice value.

Signed-off-by: Lee Duncan <leeman.duncan@gmail.com>
---
 usr/iscsi_util.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/usr/iscsi_util.c b/usr/iscsi_util.c
index 5e3420e..ac86847 100644
--- a/usr/iscsi_util.c
+++ b/usr/iscsi_util.c
@@ -60,7 +60,8 @@ int oom_adjust(void)
 	char path[ISCSI_OOM_PATH_LEN];
 	struct stat statb;
 
-	if (nice(-10) < 0)
+	errno = 0;
+	if (nice(-10) == -1 && errno != 0)
 		log_debug(1, "Could not increase process priority: %s",
 			  strerror(errno));
 
-- 
1.8.1.4

